### PR TITLE
#70 vote model

### PR DESCRIPTION
**Pull request by andrjuha01**
- [ ] @Kmalynss 
- [ ] @mrkava 
- [ ] @skcc321
Added model vote and ability to upvote and downvote the issue.
Wrote some test to cover these methods.
https://github.com/Kotany-IF070-WebUI-RoR/NewVasjuky/issues/70

### CODE REVIEW

File: app/models/vote.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :voteable, polymorphic: true
+
+  def voted?
+    result = where("user_id=#{current_user.id}")
+    result.count == 1
+  end
```
Comment (mrkava) [inline app/models/vote.rb]: why not 

where("user_id=#{current_user.id}").count == 1

we need this result variable?

File: app/views/issues/show.html.slim [hunk 2]
```diff
@@ -17,9 +17,19 @@
...
-            .followers-count
-              span Спостерігачів:
-              span.badge.pull-right #{@issue.followers_count}
+              .followers-count
+                span Спостерігачів:
+                span.badge.pull-right #{@issue.followers_count}
+        .votes-wrapper
+          .votes.pull-right
+            - if user_signed_in? && current_user.active?
+              - if @voted.empty?
```
Comment (mrkava) [inline app/views/issues/show.html.slim]: - if user_signed_in? && current_user.active?

this is repeated twice - before following block and before voting, maybe we can refactor to use this "if" just once for both follow and vote?

File: spec/models/vote_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Vote, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (mrkava) [inline spec/models/vote_spec.rb]: we need this file?

File: app/views/issues/show.html.slim (hunk 2 above)
Comment (Kmalynss) [inline app/views/issues/show.html.slim]: can you use only ```current_user.active?```?

File: app/controllers/issues_controller.rb [hunk 4]
```diff
@@ -55,6 +56,27 @@ def create
     end
   end
 
+  def upvote
+    @vote = @issue.votes.new
+    @vote.user = current_user
+
+    if @vote.save
```
Comment (Kmalynss) [inline app/controllers/issues_controller.rb]: ```ruby @vote = @issue.votes.build :user => current_user```
you can also try
```ruby if @issue.votes.create :user => current_user```

File: app/controllers/issues_controller.rb [hunk 5]
```diff
@@ -55,6 +56,27 @@ def create
...
+      redirect_to @issue, notice: 'Ви проголосували за дану проблему.'
+    else
+      redirect_to @issue, notice: 'Ви не змогли проголосувати'
+    end
+  end
+
+  def downvote
+    @vote = @issue.votes.where(user_id: current_user.id)
```
Comment (Kmalynss) [inline app/controllers/issues_controller.rb]: ```ruby @vote = @issue.votes.find_by(user_id: current_user.id)```

File: app/controllers/issues_controller.rb [hunk 6]
```diff
@@ -55,6 +56,27 @@ def create
...
+    if @vote.present?
```
Comment (Kmalynss) [inline app/controllers/issues_controller.rb]: ```ruby @issue.votes.where(user_id: current_user.id)``` always returns ActiveRecord relation object so ```ruby @vote.present? ``` will always be true.

File: app/views/issues/show.html.slim (hunk 2 above)
Comment (andrjuha01) [inline app/views/issues/show.html.slim]: @Kmalynss I don't think so, we will have the result of true every time we call "current_user.active?", we will have false only if user is banned.

File: app/models/vote.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  belongs_to :voteable, polymorphic: true
+
+  def voted?
+    where("user_id=#{current_user.id}").count == 1
```
Comment (igorkasyanchuk) [inline app/models/vote.rb]: bug. current user in model?

Comment (andrjuha01) [inline app/models/vote.rb]: @igorkasyanchuk I don't use this method, I think I need just to remove it.

File: app/controllers/issues_controller.rb [hunk 8]
```diff
@@ -55,6 +56,25 @@ def create
     end
   end
 
+  def upvote
+    @vote = @issue.votes.build :user => current_user
+    if @vote.save
+      redirect_to @issue, notice: 'Ви проголосували за дану проблему.'
```
Comment (f1nwe) [inline app/controllers/issues_controller.rb]: I thing it's bad to show notice on success.

File: app/controllers/issues_controller.rb [hunk 9]
```diff
@@ -55,6 +56,25 @@ def create
...
+    else
+      redirect_to @issue, notice: 'Ви не змогли проголосувати'
+    end
+  end
+
+  def downvote
+    @vote = @issue.votes.find_by(user_id: current_user.id)
+    if @vote.present?
+      @issue.votes.destroy(@vote)
+      redirect_to @issue, notice: 'Ви зняли голос з даної проблеми.'
```
Comment (f1nwe) [inline app/controllers/issues_controller.rb]: The same here. You should  not flash very often. For errors use alerts.

File: app/views/issues/show.html.slim [hunk 10]
```diff
@@ -17,9 +17,19 @@
...
+                = link_to upvote_issue_path(@issue), method: :post do
+                  i class="fa fa-heart-o heart" aria-hidden="true"
```
Comment (f1nwe) [inline app/views/issues/show.html.slim]: You can add class with slim syntax as "I.fa"

**Pull request merged**